server/src/application/submission.ts

```diff
@@ -860,3 +918,129 @@ export async function getSubmissionsForSurvey(surveyId: number) {
       }) as Submission,
   );
 }
+
+/**
+ * Inserts or updates the credentials so the survey submissions can be accessed using basic auth
+ * @param surveyId Survey ID
+ * @param username Username for basic auth
+ * @param password Password for basic auth
+ * @param alphanumericIncluded
+ * @param geospatialIncluded
+ * @param personalIncluded
+ * @returns Inserted row, if successful
+ */
+export async function upsertPublicationCredentials(
+  surveyId: number,
+  username: string,
+  password: string,
+  alphanumericIncluded: boolean = true,
+  geospatialIncluded: boolean = true,
+  personalIncluded: boolean = true,
+): Promise<CredentialsEntry> {
+  const row = await getDb().oneOrNone<DBCredentialsEntry>(
+    `
+    INSERT INTO
+      data.publications (
+        survey_id,
+        username,
+        password,
+        alphanumeric_included,
+        geospatial_included,
+        personal_included
+      )
+    VALUES
+      ($1, $2, crypt($3, gen_salt('bf', 8)), $4, $5, $6)
+    ON CONFLICT(survey_id)
+    DO UPDATE SET
+      username = $2,
+      password = crypt($3, gen_salt('bf', 8)),
+      alphanumeric_included = $4,
+      geospatial_included = $5,
+      personal_included = $6
+    RETURNING
+      id,
+      survey_id,
+      username,
+      alphanumeric_included,
+      geospatial_included,
+      personal_included;
+    `,
+    [
```

Comment:
Pro tip: when handling multiple DB query parameters with this DB library, you can also name them by giving them as an object and referring to them via the entry name, e.g. `$(surveyId)` instead of its array index `$1`. Array indices are fine for queries that only take one or two parameters, but they can get confusing and difficult to update when there are more parameters than that.